sql/src/main/java/io/crate/analyze/RefreshTableAnalyzer.java

```diff
@@ -35,8 +46,39 @@
     }
 
     public RefreshTableAnalyzedStatement analyze(RefreshStatement refreshStatement, Analysis analysis) {
-        Set<String> indexNames = TableAnalyzer.getIndexNames(
-            refreshStatement.tables(), schemas, analysis.parameterContext(), analysis.sessionContext().defaultSchema());
-        return new RefreshTableAnalyzedStatement(indexNames);
+        return new RefreshTableAnalyzedStatement(getIndexNames(
+            refreshStatement.tables(),
+            schemas,
+            analysis.parameterContext(),
+            analysis.sessionContext().defaultSchema()
+        ));
+    }
+
+    private static Set<String> getIndexNames(List<Table> tables,
+                                             Schemas schemas,
+                                             ParameterContext parameterContext,
+                                             @Nullable String defaultSchema) {
+        Set<String> indexNames = new HashSet<>(tables.size());
+        for (Table nodeTable : tables) {
+            TableInfo tableInfo = schemas.getTableInfo(TableIdent.of(nodeTable, defaultSchema));
+            Preconditions.checkArgument(tableInfo instanceof DocTableInfo,
+                "operation cannot be performed on system and blob tables: table '%s'",
+                tableInfo.ident().fqn());
+            if (nodeTable.partitionProperties().isEmpty()) {
+                indexNames.addAll(Arrays.asList(((DocTableInfo) tableInfo).concreteIndices()));
+            } else {
+                DocTableInfo docTableInfo = (DocTableInfo) tableInfo;
```

Comment:
Maybe something like `TableAnalyzer.filteredIndices(docTableInfo, nodeTable.paritionProperties) -> Collection<String>`.

this method could be used in both places, here and in the optimize analyzer